File: app/src/main/java/com/kevin/kggank/ui/model/MainModel.java
package com.kevin.kggank.ui.model;

import com.kevin.kggank.base.BaseModel;
import com.kevin.kggank.constants.Constants;
import com.kevin.kggank.entity.GanhuoListEntity;
import com.kevin.kggank.net.service.GanhuoService;
import com.kevin.kggank.ui.presenter.MainPresenter;
import com.kevin.kglib.rx.Compose;
import com.kevin.kglib.rx.RxUtils;

import io.reactivex.Flowable;

/**
 * Created by kevin on 16/4/15.
 */
public class MainModel extends BaseModel<MainPresenter> {

    private MainPresenter mainPresenter;

    public MainModel(MainPresenter presenter) {
        super(presenter);
        mainPresenter = presenter;
    }

    public Flowable<GanhuoListEntity> getGirls(int pageNo) {
        return RxUtils.create(GanhuoService.class)
                .getAndroidInfo(Constants.PAGE_SIZE, pageNo)
                .compose(Compose.applySchedulers());
    }

}

File: app/src/main/java/com/kevin/kggank/ui/model/MainModel.kt
package com.kevin.kggank.ui.model

import com.kevin.kggank.constants.Constants
import com.kevin.kggank.entity.GanhuoListEntity
import com.kevin.kggank.service.GanhuoService
import com.kevin.kggank.ui.presenter.MainPresenter
import com.kevin.kglib.base.LibBaseModel
import com.kevin.kglib.rx.Compose
import com.kevin.kglib.rx.RxUtils
import io.reactivex.Flowable

/**
 * Created by kevin on 16/4/15.
 */
class MainModel(private val mainPresenter: MainPresenter) : LibBaseModel<MainPresenter>(mainPresenter) {

    fun getGirls(pageNo: Int): Flowable<GanhuoListEntity> {
        return RxUtils.create(GanhuoService::class.java)
                .getAndroidInfo(Constants.PAGE_SIZE, pageNo)
                .compose(Compose.applySchedulers<GanhuoListEntity>())
    }

}
